MAUI/DataGrid/column-drag-and-drop.md

```diff
@@ -0,0 +1,226 @@
+
+{% highlight c# %}
+dataGrid.FrozenColumnCount = 2;
+
+dataGrid.QueryColumnDragging += SfDataGrid_QueryColumnDragging;
+
+private void SfDataGrid_QueryColumnDragging(object? sender, DataGridQueryColumnDraggingEventArgs e)
+{
+    //e.From returns the index of the dragged column.
+    //e.To returns the index of the current column.
+    if (e.From < 2 && e.DraggingAction == DataGridDragAction.DragEnded)
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+## Customize the appearance
+
+### Customize column drag and drop indicator color
+
+The SfDataGrid allows customizing the column drag and drop indicators using the `SfDataGrid.DefaultStyle.ColumnDraggingIndicatorLineColor` property.
+
+{% highlight xaml %}
+    <syncfusion:SfDataGrid x:Name="dataGrid"
+                           ItemsSource="{Binding OrderInfoCollection}"
+                           AllowDraggingColumn="True">
+        <syncfusion:SfDataGrid.DefaultStyle>
+            <syncfusion:DataGridStyle ColumnDraggingIndicatorLineColor="DeepPink"/>
+        </syncfusion:SfDataGrid.DefaultStyle>
+    </syncfusion:SfDataGrid>
+{% endhighlight %}
+
+<img alt="DataGrid column drag indicator color" src="Images\column-drag-and-drop\maui-datagrid-column-drag-indicator-color.png" width="450"/>
+
+### Customize drag view text color
+
+The SfDataGrid allows customizing the drag view text color using the `SfDataGrid.DefaultStyle.ColumnDragViewTextColor` property.
+
+{% highlight xaml %}
+    <syncfusion:SfDataGrid x:Name="dataGrid"
+                           ItemsSource="{Binding OrderInfoCollection}"
+                           AllowDraggingColumn="True">
+        <syncfusion:SfDataGrid.DefaultStyle>
+            <syncfusion:DataGridStyle ColumnDragViewTextColor="Red"/>
+        </syncfusion:SfDataGrid.DefaultStyle>
+    </syncfusion:SfDataGrid>
+{% endhighlight %}
+
+<img alt="DataGrid column drag view text color" src="Images\column-drag-and-drop\maui-datagrid-column-drag-view-text-color.png" width="450"/>
+
+### Customize drag view background color
+
+The SfDataGrid allows customizing the drag view background color using the `SfDataGrid.DefaultStyle.ColumnDragViewBackgroundColor` property.
+
+{% highlight xaml %}
+    <syncfusion:SfDataGrid x:Name="dataGrid"
+                           ItemsSource="{Binding OrderInfoCollection}"
+                           AllowDraggingColumn="True">
+        <syncfusion:SfDataGrid.DefaultStyle>
+            <syncfusion:DataGridStyle ColumnDragViewBackgroundColor="GreenYellow"/>
+        </syncfusion:SfDataGrid.DefaultStyle>
+    </syncfusion:SfDataGrid>
+{% endhighlight %}
+
+<img alt="DataGrid column drag view background color" src="Images\column-drag-and-drop\maui-datagrid-column-drag-view-background-color.png" width="450"/>
+
+## Cancel auto scrolling
+
+Horizontal auto-scrolling of the SfDataGrid during column drag and drop can be canceled using the `CanAutoScroll` argument of the `QueryColumnDragging` event handler.
```

Comment:
Auto-scrolling during column dragging can be disabled using the `CanAutoScroll` argument in the `QueryColumnDragging` event.

Reply:
Modified